### PR TITLE
feat: Add typings for @mertasan/tailwindcss-variables

### PR DESCRIPTION
**Pull request by LiahMartens**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/mertasan__tailwindcss-variables/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "ES2018.Promise"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "esModuleInterop": true,
```
Comment (peterblazejewicz) [inline types/mertasan__tailwindcss-variables/tsconfig.json:12]:
```suggestion
```
why do you need this?
https://github.com/DefinitelyTyped/DefinitelyTyped#esmoduleinteropallowsyntheticdefaultimports

Comment (LiamMartens) [reply types/mertasan__tailwindcss-variables/tsconfig.json:12]: @peterblazejewicz tailwindcss exports their types as `export = `

Comment (peterblazejewicz) [reply types/mertasan__tailwindcss-variables/tsconfig.json:12]: I've added a comment, just update your imports, people using this DT module could use `esModuleInterop` freely:
```ts
import * as variablesPlugin from '@mertasan/tailwindcss-variables';
import * as  variablesApi from '@mertasan/tailwindcss-variables/api';
import colorVariable = require('@mertasan/tailwindcss-variables/colorVariable');
import * as plugin from 'tailwindcss/plugin';
import type { Config } from 'tailwindcss';
```

File: types/mertasan__tailwindcss-variables/internal/PluginOptions.d.ts
Comment (peterblazejewicz) [inline types/mertasan__tailwindcss-variables/internal/PluginOptions.d.ts:1]: you can get rid of this file, as it hosts only type declaration. that can be moved to index file freely, I'll add comment how to change this

File: types/mertasan__tailwindcss-variables/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+import { Config, PluginCreator } from 'tailwindcss/types/config';
+import type { PluginOptions } from './internal/PluginOptions';
+
+declare const plugin: {
+    (options: PluginOptions): { handler: PluginCreator; config?: Config };
+    __isOptionsFunction: true;
+};
+
+export = plugin;
```
Comment (peterblazejewicz) [inline types/mertasan__tailwindcss-variables/index.d.ts:14]: you can inline types for options here:
```suggestion
import { Config, PluginCreator } from 'tailwindcss/types/config';

declare namespace plugin {
    interface PluginOptions {
        darkToRoot?: boolean;
        variablePrefix?: string;
        colorVariables?: boolean;
        extendColors?: Record<string, string>;
        forceRGB?: boolean;
        toBase?: boolean;
    }
}

declare const plugin: {
    (options: plugin.PluginOptions): { handler: PluginCreator; config?: Config };
    __isOptionsFunction: true;
};

export = plugin;
```

File: types/mertasan__tailwindcss-variables/api.d.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import type { PluginOptions } from './internal/PluginOptions';
```
Comment (peterblazejewicz) [inline types/mertasan__tailwindcss-variables/api.d.ts:1]:
```suggestion
import type { PluginOptions } from './index';
```
(if you accept inlining types)

**Review by peterblazejewicz — APPROVED** [review]
LGMT!
@LiamMartens thanks!

**Comment by LiamMartens** [comment]
Ready to merge

**Pull request merged**